From: Peter Zijlstra <peterz@infradead.org>
Date: Tue, 12 Aug 2025 12:39:05 +0200
Subject: perf: Remove redundant aux_unlock label
Git-commit: 41b80e1d74bdef5e48ea63d186244b9f6f82a4da
Patch-mainline: v6.18-rc1
References: perf-events-v6.19-update

unlock and aux_unlock are now identical, remove the aux_unlock one.

Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Reviewed-by: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Link: https://lore.kernel.org/r/20250812104019.131293512@infradead.org

Signed-off-by: Tony Jones <tonyj@suse.de>
---
 kernel/events/core.c | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/kernel/events/core.c b/kernel/events/core.c
index dfe09b033273..89fb069913d0 100644
--- a/kernel/events/core.c
+++ b/kernel/events/core.c
@@ -7098,7 +7098,7 @@ static int perf_mmap(struct file *file, struct vm_area_struct *vma)
 
 		rb = event->rb;
 		if (!rb)
-			goto aux_unlock;
+			goto unlock;
 
 		aux_mutex = &rb->aux_mutex;
 		mutex_lock(aux_mutex);
@@ -7107,27 +7107,27 @@ static int perf_mmap(struct file *file, struct vm_area_struct *vma)
 		aux_size = READ_ONCE(rb->user_page->aux_size);
 
 		if (aux_offset < perf_data_size(rb) + PAGE_SIZE)
-			goto aux_unlock;
+			goto unlock;
 
 		if (aux_offset != vma->vm_pgoff << PAGE_SHIFT)
-			goto aux_unlock;
+			goto unlock;
 
 		/* already mapped with a different offset */
 		if (rb_has_aux(rb) && rb->aux_pgoff != vma->vm_pgoff)
-			goto aux_unlock;
+			goto unlock;
 
 		if (aux_size != nr_pages * PAGE_SIZE)
-			goto aux_unlock;
+			goto unlock;
 
 		/* already mapped with a different size */
 		if (rb_has_aux(rb) && rb->aux_nr_pages != nr_pages)
-			goto aux_unlock;
+			goto unlock;
 
 		if (!is_power_of_2(nr_pages))
-			goto aux_unlock;
+			goto unlock;
 
 		if (!atomic_inc_not_zero(&rb->mmap_count))
-			goto aux_unlock;
+			goto unlock;
 
 		if (rb_has_aux(rb)) {
 			atomic_inc(&rb->aux_mmap_count);
@@ -7161,7 +7161,6 @@ static int perf_mmap(struct file *file, struct vm_area_struct *vma)
 	}
 
 unlock:
-aux_unlock:
 	if (aux_mutex)
 		mutex_unlock(aux_mutex);
 	mutex_unlock(&event->mmap_mutex);

